### PR TITLE
Make HTTP origin value an instance member

### PR DESCRIPTION
**Pull request by JunielKatarn**
## Description
Sets HTTP origin URL per filter instance.

### Type of Change
- Bug fix (non-breaking change which fixes an issue)

### Why
A crash can be caused by an inconsistent state when the host app is being shut down and an HTTP request has been scheduled from the JS layer.

This affects the static member `OriginPolicyHttpFilter::s_origin`.

### What
Make the HTTP origin an instance member of `OriginPolicyHttpFilter` so the value will be bound to the filter's life time.

In most cases, this won't add significant memory overhead because the filter's instance (and its members) is a singleton for the owning resource, module, and React instance.

## Testing
Updated tests `ValidatePreflightResponseHeadersCaseMismatchSucceeds` and `ValidatePreflightResponseMainAndContentHeadersSucceeds`.

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/react-native-windows/pull/12821)

### CODE REVIEW

File: vnext/Shared/Networking/WinRTHttpResource.cpp [hunk 1]
```diff
@@ -653,8 +652,7 @@ void WinRTHttpResource::AddResponseHandler(shared_ptr<IResponseHandler> response
     client = HttpClient{redirFilter};
   } else {
     auto globalOrigin = GetRuntimeOptionString("Http.GlobalOrigin");
```
Comment (acoates-ms) [inline vnext/Shared/Networking/WinRTHttpResource.cpp:654]: We should add a public PropertyId, or a setter for ReactPropertyBag/ReactInstanceSettings to allow setting this option on a per instance basis.  If someone wanted to have different origins for different RN instances there would still be a synchronization issue here.

We still need the "legacy" GetRuntimeOptionString option for Office's non-abi code paths, but for public use this should all be done using ReactInstanceSettings.

Comment (JunielKatarn) [reply vnext/Shared/Networking/WinRTHttpResource.cpp:654]: Can that be addressed in a follow-up PR? (I could use that to re-factor other property ID declarations that are done inefficiently).

This fix is required ASAP.

Comment (acoates-ms) [reply vnext/Shared/Networking/WinRTHttpResource.cpp:654]: Sure

**Review by acoates-ms — APPROVED** [review]

File: vnext/Desktop.UnitTests/OriginPolicyHttpFilterTest.cpp [hunk 2]
```diff
@@ -280,23 +280,20 @@ TEST_CLASS (OriginPolicyHttpFilterTest) {
     // Should implicitly set Content-Length and Content-Type
     request.Content(HttpStringContent{L"PreflightContent"});
 
-    auto filter = winrt::make<OriginPolicyHttpFilter>(mockFilter);
+    auto filter = winrt::make<OriginPolicyHttpFilter>("http://somehost", mockFilter);
     auto opFilter = filter.as<OriginPolicyHttpFilter>();
```
Comment (NikoAri) [inline vnext/Desktop.UnitTests/OriginPolicyHttpFilterTest.cpp:284]: That's not good to have this, someone can take over this Url.
Please use microsoft Urls at least, or ask Danny for best practice.

Comment (JunielKatarn) [reply vnext/Desktop.UnitTests/OriginPolicyHttpFilterTest.cpp:284]: These are unit tests.
No actual network call his being made to any of the indicated URLs.

They are used only for address-handling validations.

**Pull request merged**